multiqc/modules/dragen/coverage_per_contig.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import itertools
+import math
+import re
+from collections import OrderedDict, defaultdict
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.plots import linegraph, bargraph
+
+# Initialise the logger
+import logging
+log = logging.getLogger(__name__)
+
+
+class DragenCoveragePerContig(BaseMultiqcModule):
+    def add_coverage_per_contig(self):
+        perchrom_data_by_phenotype_by_sample = defaultdict(dict)
+
+        for f in self.find_log_files('dragen/wgs_contig_mean_cov'):
+            perchrom_data_by_phenotype = parse_wgs_contig_mean_cov(f)
+            if f['s_name'] in perchrom_data_by_phenotype_by_sample:
+                log.debug('Duplicate sample name found! Overwriting: {}'.format(f['s_name']))
+            self.add_data_source(f, section='stats')
+            perchrom_data_by_phenotype_by_sample[f['s_name']].update(perchrom_data_by_phenotype)
+
+        # Filter to strip out ignored sample names:
+        perchrom_data_by_phenotype_by_sample = self.ignore_samples(perchrom_data_by_phenotype_by_sample)
+
+        # Merge tumor and normal data:
+        perchrom_data_by_sample = defaultdict(dict)
+        for sn in perchrom_data_by_phenotype_by_sample:
+            for phenotype in perchrom_data_by_phenotype_by_sample[sn]:
+                new_sn = sn
+                if phenotype == 'normal':
+                    new_sn = sn + '_normal'
+                perchrom_data_by_sample[new_sn] = perchrom_data_by_phenotype_by_sample[sn][phenotype]
+
+        if not perchrom_data_by_sample:
+            return
+        log.info('Found DRAGEN per-contig coverage histogram for {} DRAGEN output prefixes'.format(
+            len(perchrom_data_by_sample)))
+
+        main_contigs_by_sample = {sn: data[0] for sn, data in perchrom_data_by_sample.items()}
+        other_contigs_by_sample = {sn: data[1] for sn, data in perchrom_data_by_sample.items()}
+
+        self.add_section(
+            name='Coverage per contig',
+            anchor='dragen-coverage-per-contig',
+            description='Average coverage per contig or chromosome. '
+                        'Calculated as the number of bases (excluding duplicate marked reads, reads '
+                        'with MAPQ=0, and clipped bases), divided by the length of the contig or '
+                        '(if a target bed is used) the total length of the target region spanning that contig.',
```

Comment:
I've started using multi-line strings a lot more for these, I find them quite a bit easier to read:

```suggestion
            description='''
            Average coverage per contig or chromosome.
            Calculated as the number of bases (excluding duplicate marked reads, reads
            with `MAPQ=0`, and clipped bases), divided by the length of the contig or
            _(if a target bed is used)_ the total length of the target region spanning that contig.
            ''',
```

Reply:
Normally I avoid multiline strings because you need to deal with all the whitespace and empty lines from indentation. But here it makes total sense, it gets converted into html so all whitespace is stripped.

Reply:
Yup - even better, I remove the whitespace at the start of the block for you - as long as it's even:

https://github.com/ewels/MultiQC/blob/c5d3c1d8a0b865366aa50d8ef94a18727150c5a6/multiqc/modules/base_module.py#L167